viewing_party/party.py

```diff
@@ -1,23 +1,162 @@
+# Testing connection
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+
+    if title == None or genre == None or rating == None:
+        return None
+
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title): 
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            add_to_watched(user_data, movie)
+            user_data["watchlist"].remove(movie)
+    
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+
+    rating_lst = []
+    for movie in user_data["watched"]:
+        rating_lst.append(movie["rating"])
+    
+    if len(rating_lst) == 0: 
+        return 0
+
+    return sum(rating_lst)/len(rating_lst)
 
+def get_most_watched_genre(user_data):
+    
+    watched_lst = user_data["watched"]
+    genre_freq = {}
+    most_watched_count = 0
+    most_watched_genre = ""
+
+    # Check empty watched list
+    if not watched_lst:
+        return None
+    
+    for movie in watched_lst:
+        genre = movie["genre"]
+
+        # Count genre frequency
+        if genre not in genre_freq.keys():

```

Comment:
The 'in' operator also works for keys in a dictionary, so it is not necessary to grab the keys as a list.  Also, the `in` operator on a list is O(n), but using the `in` operator on a dictionary is O(1).
```suggestion
        if genre not in genre_freq:
```